### PR TITLE
feat(dot/parachain/backing): handle `GetBackedCandidatesMessage` incoming overseer message

### LINKED ISSUE #3504: feat(dot/parachain/backing): handle `GetBackedCandidates` incoming overseer message
## Issue summary

- Can only be tackled after https://github.com/ChainSafe/gossamer/pull/3497 is merged
- Here in Gossamer
https://github.com/ChainSafe/gossamer/blob/8f10724821e8d3a4ae502e0c5e8691bd98cdd12f/dot/parachain/backing/candidate_backing.go#L101-L102

## Other information and links

Reference code in polkadot-sdk
https://github.com/paritytech/polkadot-sdk/blob/769bdd3ff33a291cbc70a800a3830638467e42a2/polkadot/node/core/backing/src/lib.rs#L754-L755

### PR DESCRIPTION
**Pull request by axaysagathiya**
## Note
need to merge ONLY AFTER #3682  gets merged

## Tests

```sh
go test -tags integration github.com/ChainSafe/gossamer
```

## Issues

Closes #3504

## Primary Reviewer

@kishansagathiya @EclesioMeloJunior

### CODE REVIEW

File: dot/parachain/backing/statement_table.go
Comment (kishansagathiya) [inline dot/parachain/backing/statement_table.go:12]: can you check and tell why `parachaintypes.CandidateHash` to pointer?

**Review by kishansagathiya — COMMENTED** [review]
Looks good so far! Will approve after dependent PRs are merged!

File: dot/parachain/backing/statement_table.go
Comment (axaysagathiya) [reply dot/parachain/backing/statement_table.go:11]: We will not modify the value of this variable. We don't need a pointer here.
What's your POV? Should we use a pointer?

**Review by kishansagathiya — APPROVED** [review]

**Review by EclesioMeloJunior — APPROVED** [review]

**Pull request merged**
